rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/SymbolTable.kt

```diff
@@ -51,27 +51,14 @@ class SymbolTable {
         throw IllegalArgumentException("Cannot find searchedValued for $dataName")
     }
 
-    fun dataDefinitionByName(dataName: String) =
+    override fun dataDefinitionByName(dataName: String) =
             values.keys.firstOrNull { it.name.equals(dataName, ignoreCase = true) }
 
-    operator fun set(data: AbstractDataDefinition, value: Value) {
+    override operator fun set(data: AbstractDataDefinition, value: Value): Value? {
         require(!(data !in this && data.name in this)) {
             "This data definition would conflict with an existing data definition with the same name. This data definition: $data. Existing data definition: ${this[data.name]}"
         }
         require(data.type.canBeAssigned(value))
-        values[data] = value.forType(data.type)
+        return values.put(data, value.forType(data.type))
```

Comment:
`values[data] =` is a bit more kotlinish

Reply:
I need to return the old value but "values[data]" is not expression.
Have you some suggestion?

Reply:
You're right